backend/server.js

```diff
@@ -15,13 +37,66 @@ const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+const authenticateUser = async (req, res, next) => {
+try {  
+  const user = await User.findOne({ 
+    accessToken: req.header('Authorization') 
+  })
+  if (user) {
+    req.user = user
+    next()
+  } else {
+    res.status(401).json({ loggedOut: true, message: 'Please try login again' })
+  }
+} catch (err) {
+  res
+    .status(403)
+    .json({ message: 'access token missing or wrong', errors: err.errors })
+}
 
 // Start defining your routes here
 app.get('/', (req, res) => {
   res.send('Hello world')
+ });
+
+ app.post('/register', async (req, res) => {
+  try {
+    const { name, email, password} = req.body;
+    const user = new User({ name, email, password: bcrypt.hashSync(password) });
+    const saved = await user.save();
+    res.status(201).json(saved);
+  } catch (err) {
+    res.status(400).json({ message: 'could not save user', error: err.errors })
+  }  
+});
+
+ app.get('/users/:id', authenticateUser)
+ app.get('/users/:id', (req, res) => {
+   try {
+     res.status(201).json(req.user)
+   } catch (err) {
+     res.status(400).json({message: 'could not save user', errors: err.message})
+   }
+ })
+
+// login user
+app.post('/sessions', async (req, res) => {
+  try {
+    const { name, password } = req.body
+    const user = await User.findOne({ name })
+    if (user && bcrypt.compareSync(password, user.password)) {
+      res.status(201).json({ useId: user._id, accessToken: user.accessToken })
+    } else {
+      res.json({ message: "wrong username or paaaword" })
+    }
+  } catch (err) {
+    res.status(400).json({ message: 'could not find user', errors: err.errors })
+  }
 })
+}  
```

Comment:
This extra `}` is actually closing the function on line https://github.com/Technigo/project-auth/pull/12/files#diff-5784859bc20a9558df9dfce578b2f673R43

Line 43: ` const authenticateUser = async (req, res, next) => { `

Which means ALL of your endpoints are not defined until this function is called.  

This is where formatting can be SOO helpful.  You could see that the functions weren't exactly on the left edge of the editor, so they were inside of a different code block.  

Once this was fixed.  I was able to create a user.

Reply:
Thank you! I understand that I have a } on the wrong line. But where is it supposed to be? line 59? :)

Reply:
Or was it some other line? :)

Reply:
Yeah, it was supposed to be on 59 to close that function.  Once you change that, the other endpoints are defined outside of that function and work properly 👍